### PR TITLE
docs(pipelines): adds index page for overall pipelines guide

### PR DESCRIPTION
**Pull request by dorbin**

### CODE REVIEW

**Comment by dorbin** [comment]
#1027

**Review by skim1420 — APPROVED** [review]
LGTM. It's worthwhile to do a run through of other 404s that may exist for pages that are navigable via clicking up a page from an existing page.

File: reference/pipeline/index.md [hunk 1]
```diff
@@ -0,0 +1,9 @@
+---
+layout: single
+title:  "Spinnaker Pipeline Reference"
+sidebar:
+  nav: reference
+---
+
+* [Pipeline Expression Reference](/reference/pipeline/expressions/)
+* [Pipeline Stages](/reference/pipeline/stages/)
```
Comment (avh121) [inline reference/pipeline/index.md:9]: nit: do we want to say "Pipeline Stages Reference" for consistency here? I don't have strong feelings, but I think it looks a bit funny to have "Pipeline Expression Reference" and then just "Pipeline Stages."

**Review by avh121 — APPROVED** [review]

**Review by lwander — APPROVED** [review]

File: reference/pipeline/index.md (hunk 1 above)
Comment (dorbin) [reply reference/pipeline/index.md:9]: I was matching the existing titles on the target pages ;)

**Pull request merged**